### PR TITLE
feat: attempt to add source ids to quotas

### PR DESCRIPTION
**Pull request by WillKocur**
Adds source_ids to quotas and suspensions

Seems to work now, although not sure if it's the most efficient, nor if there are some unnecessary bits in the code.

Also, is there any way that this could affect other sites?

### CODE REVIEW

File: app/controllers/api/v1/taxon_concepts_controller.rb [hunk 1]
```diff
@@ -22,7 +22,7 @@ def show
     @taxon_concept = TaxonConcept.
       includes(:common_names => :language,
                :distributions => :geo_entity,
-               :quotas => :geo_entity,
+               :quotas => [:geo_entity, :sources],
                :cites_suspensions => :geo_entity).
       includes(:taxonomy).find(params[:id])
```
Comment (WillKocur) [inline app/controllers/api/v1/taxon_concepts_controller.rb:27]: Not clear to me why this includes is necessary...

**Review by lucacug — COMMENTED** [review]
This looks good, the SQL is great stuff, I'm not 100% convinced by the correlated query, maybe there is a way to avoid that and do a simple join with the trade_restriction but this is working well so I'm happy to keep it as it is.
I just want to do a few more tests with different apps to see if anything breaks, it shouldn't as you are only including a new field to the API, so worst case scenario, other apps will be blind to that but worth it double check

**Comment by WillKocur** [comment]
> This looks good, the SQL is great stuff, I'm not 100% convinced by the correlated query, maybe there is a way to avoid that and do a simple join with the trade_restriction but this is working well so I'm happy to keep it as it is. I just want to do a few more tests with different apps to see if anything breaks, it shouldn't as you are only including a new field to the API, so worst case scenario, other apps will be blind to that but worth it double check

Yeah, I struggled to find another way to do this, given that I needed to aggregate the source ids. Although I've not really worked with SQL properly before, so could easily have missed something!

File: db/migrate/20221014141105_add_source_ids_to_api_cites_suspensions_view.rb [hunk 2]
```diff
@@ -0,0 +1,6 @@
+class AddSourceIdsToApiCitesSuspensionsView < ActiveRecord::Migration
+  def change
+    execute "DROP VIEW IF EXISTS api_cites_suspensions_view"
+    execute "CREATE VIEW api_cites_suspensions_view AS #{view_sql('20221014151355', 'api_cites_suspensions_view')}"
+  end
```
[the remark this answers is not in the document]
Comment (WillKocur) [reply db/migrate/20221014141105_add_source_ids_to_api_cites_suspensions_view.rb]: Ah yes, good spot - done!

**Pull request merged**